azurerm/internal/services/securitycenter/server_vulnerability_assessment_resource.go

```diff
@@ -0,0 +1,233 @@
+	readError := resourceServerVulnerabilityAssessmentRead(d, meta)
+	for retry := 1; d.Id() == "" && retry < 11; retry++ {
+		time.Sleep(time.Duration(retry) * time.Second)
+		readError = resourceServerVulnerabilityAssessmentRead(d, meta)
+	}
+	return readError
+}
+
+func resourceServerVulnerabilityAssessmentRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).SecurityCenter.ServerVulnerabilityAssessmentClient
+	ctx, cancel := timeouts.ForRead(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	targetId, err := parseTargetId(d)
+	if err != nil {
+		return err
+	}
+
+	resp, err := client.Get(ctx, targetId.ResourceGroup, targetId.Provider, targetId.ResourceType, targetId.Name)
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			log.Printf("Server Vulnerability Assessment was not found for %q: %+v", targetId.Id, err)
+			d.SetId("")
+			return nil
+		}
+
+		return fmt.Errorf("retrieving Server Vulnerability Assessment status for %q: %+v", targetId.Id, err)
+	}
+
+	// Update state
+	d.SetId(*resp.ID)
+
+	// Handle import
+	switch {
+	case targetId.Provider == virtualMachineProvider && d.Get("virtual_machine_id") == "":
+		_ = d.Set("virtual_machine_id", targetId.Id)
+	case targetId.Provider == hybridMachineProvider && d.Get("hybrid_machine_id") == "":
+		_ = d.Set("hybrid_machine_id", targetId.Id)
+	default:
+	}
+
+	return nil
+}
+
+func resourceServerVulnerabilityAssessmentDelete(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).SecurityCenter.ServerVulnerabilityAssessmentClient
+	ctx, cancel := timeouts.ForDelete(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	targetId, err := parseTargetId(d)
+	if err != nil {
+		return err
+	}
+
+	// Can not delete if still in provisioning state. Wait for it to complete.
+	for retry := 1; retry <= 20; retry++ {
+		response, err := client.Get(ctx, targetId.ResourceGroup, targetId.Provider, targetId.ResourceType, targetId.Name)
+		if err != nil || response.ProvisioningState != security.ProvisioningState1Provisioning {
+			break
+		}
+		time.Sleep(time.Duration(retry) * time.Second)
+	}
+
+	response, err := client.Delete(ctx, targetId.ResourceGroup, targetId.Provider, targetId.ResourceType, targetId.Name)
+	if err != nil {
+		// https://github.com/Azure/azure-sdk-for-go/issues/14120 the API report error even if delete was successful
+		if utils.ResponseWasStatusCode(response, 202) {
+			return nil
+		} else {
+			return fmt.Errorf("removing Server Vulnerability Assessment for %q: %+v\n", targetId.Id, err)
+		}
+	} else {
+		return nil
+	}
+}
+
+type TargetId struct {
+	Id             string
+	SubscriptionId string
+	ResourceGroup  string
+	Provider       string
+	ResourceType   string
+	Name           string
+}
+
+func parseTargetId(d *schema.ResourceData) (*TargetId, error) {
+	rawVirtualMachineId := d.Get("virtual_machine_id").(string)
+	rawHybridMachineId := d.Get("hybrid_machine_id").(string)
```

Comment:
Does it make sense to parse it by judging whether the "target_machine_id" exists the resource id segment "Microsoft.HybridCompute/machines" after merged these two properties to one property "target_machine_id"?

Reply:
Yes, depending on if we merge the properties or not.